tools/run-integration-test.sh

```diff
@@ -0,0 +1,186 @@
+#!/bin/bash
+# OSDFIR Infrastructure integration tests
+# This script can be used to test the integration between dfTimewolf, Timesketch, and Turbinia
+# Requirements:
+# - have 'kubectl', 'jq' and 'expect' packages installed
+# - have the OSDFIR Infrastructure Helm chart deployed and are authenticated to the GKE cluster
+# - have the 'dfTimewolf', 'turbinia-client', and 'timesketch' CLI's installed
+
+set -o posix
+set -e
+
+RELEASE="test"
+DISK="disk-1"
+SKETCH_ID="1"
+FAILED=0
+DATE=$(date -I)
+
+if [ $# -ne  2 ]
+then
+  echo "Not enough arguments supplied, please provide GCP project and zone"
+  echo "$0 [PROJECT] [ZONE]"
+  exit 1
+fi
+
+GCP_PROJECT="$1"
+GCP_ZONE="$2"
+
+echo -n "Started at "
+date -Iseconds
+
+# Forward k8s services
+echo "Forwarding k8s $RELEASE services"
+kubectl --namespace default port-forward service/$RELEASE-timesketch 5000:5000 > /dev/null 2>&1 &
+kubectl --namespace default port-forward service/$RELEASE-turbinia 8000:8000  > /dev/null 2>&1 &
+
+# Back up existing Timesketch configs else script will attempt to connect to wrong Timesketch instance
+if  [ -f ~/.timesketchrc ] && [ -f ~/.timesketch.token ] 
+then
+    echo "Backing up existing Timesketch configs to ~/.timesketchrc.$DATE and ~/.timesketch.token.$DATE"
+    mv ~/.timesketchrc ~/.timesketchrc.$DATE
+    mv ~/.timesketch.token ~/.timesketch.token.$DATE
+fi
+
+echo "Generating Timesketch config..."
+TS_SECRET=$(kubectl get secret --namespace default $RELEASE-timesketch-secret -o jsonpath="{.data.timesketch-user}" | base64 -d)
+if ! [ -f ~/.timesketchrc ] && ! [ -f ~/.timesketch.token ]; then
+    expect <<END_EXPECT
+    spawn timesketch config
+    expect "What is the value for <host_uri>"
+    send "http://127.0.0.1:5000\r"
+    expect "What is the value for <auth_mode>"
+    send "userpass\r"
+    expect "What is the value for <username>"
+    send "timesketch\r"
+    expect "Password for user timesketch"
+    send "$TS_SECRET\r"
+    expect eof
+END_EXPECT
+fi
+
+# Back up existing Turbinia config else script will attempt to connect to wrong Turbinia instance
+if  [ -f ~/.turbinia_api_config.json ]
+then
+  echo "Backing up existing Turbinia config to ~/.turbinia_api_config.json.$DATE"
+  mv ~/.turbinia_api_config.json ~/.turbinia_api_config.json.$DATE
+fi
+
+# Replace Turbinia config with test config
+echo "Writing turbinia config to $HOME/.turbinia_api_config.json..."
+cat > $HOME/.turbinia_api_config.json <<EOL
+{
+	"default": {
+		"description": "Turbinia client test config",
+		"API_SERVER_ADDRESS": "http://127.0.0.1",
+		"API_SERVER_PORT": 8000,
+		"API_AUTHENTICATION_ENABLED": false,
+		"CLIENT_SECRETS_FILENAME": ".client_secrets.json",
+		"CREDENTIALS_FILENAME": ".credentials_default.json"
+	}
+}
+EOL
+
+# Run dfTimewolf recipe
```

Comment:
In the longer run, let's make it easy to run multiple different recipes

Reply:
Yeah agreed, once a solution comes to mind on how to best do that ill update in another pass so we can re-use for other tests :)